mario/tools/tabletransform.py

```diff
@@ -232,3 +232,48 @@ def SUT_to_IOT(instance, method):
     rename_index(matrices["baseline"])
 
     return matrices, indeces, units
+
+
+def ISARD_TO_CHENERY_MOSES(instance,scenario):
+
+    regions = instance.get_index(_MASTER_INDEX['r'])
+    commodities = instance.get_index(_MASTER_INDEX['c'])
+    sN = slice(None)
+
+    U_isard = instance.get_data(['U'], scenarios=[scenario])[scenario][0]
+    Y_isard = instance.get_data(['Y'], scenarios=[scenario])[scenario][0]
+    s_isard = instance.get_data(['s'], scenarios=[scenario])[scenario][0]
+
+    domestic_use = pd.DataFrame(0, index=U_isard.index, columns=regions)
+    for region in regions:
+        df = pd.DataFrame((U_isard.loc[:,(region,sN,sN)].sum(axis=1) + Y_isard.loc[(sN,_MASTER_INDEX['c'],sN),(region,sN,sN)].sum(axis=1)).values,index=U_isard.index,columns=[region])
+        domestic_use.update(df)
+    
+    U_chenery = pd.DataFrame(0, index=U_isard.index, columns=U_isard.columns)
+    Y_chenery = pd.DataFrame(0, index=Y_isard.index, columns=Y_isard.columns)
+    S_chenery = pd.DataFrame(0, index=s_isard.index, columns=s_isard.columns)
+    for region in regions:
+        domestic_U = U_isard.loc[:,(region,sN,sN)].groupby(level=2).sum()
+        domestic_U.index = U_isard.loc[(region,sN,sN),:].index
+        U_chenery.loc[(region,sN,sN),(region,sN,sN)] = domestic_U.values
+
+        domestic_Y = Y_isard.loc[(sN,_MASTER_INDEX['c'],sN),(region,sN,sN)].groupby(level=2).sum()
+        domestic_Y.index = Y_isard.loc[(region,_MASTER_INDEX['c'],sN),:].index
+        Y_chenery.loc[(region,_MASTER_INDEX['c'],sN),(region,sN,sN)] = domestic_Y.values
+
+        for region_2 in regions:
+            for commodity in commodities:
```

Comment:
is it possible to avoid the loops throguh vectoral operations of pandas?

Reply:
I changed function to avoid looping over commodities. Tested on Exiobase and timing is less than half. Consider most of the time now is taken to calculate "z" which is always necessary.